webapp/app/js/compare.js

```diff
@@ -0,0 +1,494 @@
+                total += perfData[i].geomean;
+              }
+
+              var avg = total / geomeans.length;
+              var sigma = stddev(geomeans, avg);
+              rawResultsMap[data.series_signature] = {'geomean': avg, 'variation': 2*sigma, 'runs': geomeans.length};
+            });
+
+
+            //ok, now get the new revision
+            signatureURL = thServiceDomain + '/api/project/' + $scope.newProject + 
+              '/performance-data/0/get_performance_series_summary/?interval=' +
+              $scope.timeRange;
+
+            $http.get(signatureURL).then(function(response) {
+              var data = response.data;
+              var new_seriesList = [];
+              var new_selectedSignatures = [];
+              var new_selectedSignaturesCount = 0;
+              Object.keys(data).forEach(function(signature) {
+                var seriesSummary = getSeriesSummary(signature,
+                                                     data[signature],
+                                                     optionCollectionMap,
+                                                     $scope.pgo,
+                                                     $scope.e10s);
+
+                if (seriesSummary == {}) {
+                  return;
+                }
+
+                if (seriesSummary.signature !== undefined && new_selectedSignaturesCount < 20) {
+                  new_seriesList.push(seriesSummary);
+                  new_selectedSignatures.push(seriesSummary.signature);
+                  new_selectedSignaturesCount++;
+                }
+              });
+
+              // find summary results for all tests/platforms for the original rev
+              uri2 = thServiceDomain + '/api/project/' +
+                     $scope.newProject + '/performance-data/0/' +
+                     'get_performance_data/?interval_seconds=' + $scope.timeRange;
+              new_selectedSignatures.forEach(function(signature) {
+                uri2 += ('&signatures=' + signature);
+              });
+
+              // TODO: figure how how to reduce these maps
+              var new_rawResultsMap = {};
+
+              $http.get(uri2).then(function(response) {
+                response.data.forEach(function(data) {
+
+                  var perfData = data.blob;
+                  var geomeans = [];
+                  var total = 0;
+                  for (var i=0; i < perfData.length; i++) {
+                    if (perfData[i].result_set_id != $scope.newResultSetID) {
+                      continue;
+                    }
+
+                    geomeans.push(perfData[i].geomean);
+                    total += perfData[i].geomean;
+                  }
+
+                  var avg = total / geomeans.length;
+                  var sigma = stddev(geomeans, avg);
+                  new_rawResultsMap[data.series_signature] = {'geomean': avg, 'variation': 2*sigma, 'runs': geomeans.length};
+                });
```

Comment:
You really need to factor this duplicated code to get results for prev/next revisions into a common function and call it twice. :)